src/lib/services/feature-toggle-service.ts

```diff
@@ -377,17 +377,53 @@ class FeatureToggleService {
             constraints: featureStrategy.constraints || [],
             parameters: featureStrategy.parameters,
             segments: segments.map((segment) => segment.id) ?? [],
+            sortOrder: featureStrategy.sortOrder,
         };
     }
 
     async updateStrategiesSortOrder(
         featureName: string,
+        environment: string,
+        project: string,
+        createdBy: string,
         sortOrders: SetStrategySortOrderSchema,
     ): Promise<Saved<any>> {
         await Promise.all(
-            sortOrders.map(async ({ id, sortOrder }) =>
-                this.featureStrategiesStore.updateSortOrder(id, sortOrder),
-            ),
+            sortOrders.map(async ({ id, sortOrder }) => {
+                const strategyToUpdate =
+                    await this.featureStrategiesStore.getStrategyById(id);
+                await this.featureStrategiesStore.updateSortOrder(
+                    id,
+                    sortOrder,
+                );
+                const updatedStrategy =
+                    await this.featureStrategiesStore.getStrategyById(id);
+
+                const tags = await this.tagStore.getAllTagsForFeature(
+                    featureName,
+                );
+                const segments = await this.segmentService.getByStrategy(
```

Comment:
do we need to fetch those segments? i think we only ned data and preData for diffing. Since we know only the order change maybe we can have segments: [] in both variants?

Reply:
I guess the only problem would be data in the event log showing incorrect value so probably it's not a good idea